packages/flutter/test/cupertino/scrollbar_paint_test.dart

```diff
@@ -79,21 +79,8 @@ void main() {
     await gesture.moveBy(Offset(-_kGestureOffset.dx, -_kGestureOffset.dy));
     await tester.pump();
     await tester.pump(const Duration(milliseconds: 500));
-
-    expect(find.byType(CupertinoScrollbar), paints..rrect(
-      color: _kScrollbarColor,
-      rrect: RRect.fromRectAndRadius(
-        const Rect.fromLTWH(
-          800.0 - 3 - 3, // Screen width - margin - thickness.
-          44 + 20 + 3.0, // nav bar height + top margin
-          3, // Thickness.
-          // Fraction visible * (viewport size - padding - margin)
-          // where Fraction visible = (viewport size - padding) / content size
-          (600.0 - 34 - 44 - 20) / 4000.0 * (600.0 - 2 * 3 - 34 - 44 - 20),
-        ),
-        _kScrollbarRadius,
-      ),
-    ));
+    expect(tester.getRect(find.byType(CupertinoScrollbar)),
+        rectMoreOrLessEquals(const Rect.fromLTWH(0, 0, 800, 600), epsilon: 1)); // (800, 600) is viewport size
```

Comment:
This does not look right, does the scrollbar overlap the navigation bar now?

Reply:
What iOS does is that, if the scrollbar is full screen, then the scrollbar follows the safe area. If the scrollbar is attached to, say, a card, then the scrollbar shouldn't have any paddings.

So the problem is, either we have a way for the scrollbar to automatically detect what it is attached to, or its parent should be responsible to decide whether the scrollbar should use the view padding.

For the first option, if we decide that it's the parent's responsibility to warp itself in `SafeArea`, then we don't need this PR at all. However I don't think this is a good way, because we have way more "non-fullscreen" widgets than fullscreen widgets.

Therefore I'm basically proposing that the scrollbar should by default assume it's attached to a non-fullscreen widget, and the parent should explicitly apply view padding to the scrollbar if the parent knows itself is fullscreen.

Reply:
> the parent should explicitly apply view padding to the scrollbar if the parent knows itself is fullscreen.

Wouldn't this be a breaking change then?

> yes, the scrollbar overlaps the navigation bar.

As a user I would not expect this, especially since on some platforms, the scrollbar is added automatically and the user does not have as much control over it.

Reply:
Alternatively we can also add a paragraph to the doc to notify the user of the default padding, and suggest them to either use a SafeArea or pass in a zero padding in non-fullscreen cases. What do you think?

Reply:
> Alternatively we can also add a paragraph to the doc to notify the user of the default padding, and suggest them to either use a SafeArea or pass in a zero padding in non-fullscreen cases. What do you think?

I was thinking if we provide padding from top equal to `kToolBarHeight` and from bottom also `kToolBarHeight` only when bottom navigation bar and app bar is enabled for that Widget, else padding remains 0.
I might be wrong this is just an idea which i think might work.

Reply:
Users already expect that to work though without having to manually configure it. 🤔

Reply:
> > Alternatively we can also add a paragraph to the doc to notify the user of the default padding, and suggest them to either use a SafeArea or pass in a zero padding in non-fullscreen cases. What do you think?
> 
> 
> 
> I was thinking if we provide padding from top equal to `kToolBarHeight` and from bottom also `kToolBarHeight` only when bottom navigation bar and app bar is enabled for that Widget, else padding remains 0.
> 
> I might be wrong this is just an idea which i think might work.


I don't quite get your proposal but I dont't think that's going to work. It sounds to me that they'll make it only work for situations that the view paddings come from app bars, which doesn't solve the case that I encountered (using scrollbars on a card) either...

Reply:
> Users already expect that to work though without having to manually configure it. 🤔 

Another more complicated solution is to detect the boundary of the Scrollbar widget in terms of screen space, and calculate the padding it needs to use dynamically. For example, if there's scrollbar is on a card that is 10 px within the edge of the screen, then the padding of the scrollbar will be viewPadding-10.

Reply:
I wonder if we should take a step back, maybe I have lost sight of the root issue. The scrollbar determines its bounds based on the child it wraps, which is usually the viewport (the window scrolling is happening in).

When I look at the video in the OP of the issue, I think oh yeah that padding should totally be removed so the notch of the phone does not affect the scrollbar in the dialog. Then I realized, shouldn't the dialog handle that then? The dialog has consumed the view padding. Should the dialog remove the padding then? Scaffold does this, it removes the padding as it's children consume it, like with AppBar and BottomNavigationBar.

Reply:
Yeah, I'm mostly thinking that the current default is counterintuitive, since I would consider full screen scrollbars rarer than non fullscreen ones. But I agree that we don't have to change it, but if we don't I think we should improve the doc.

Reply:
@Piinks How do you think we should move on? Should we improve the doc instead (or should we at all?)

Reply:
> How do you think we should move on? 

If we can update this change to not have the overlap of the navigation bar by default, that would be a path forward. Right now as it is, it is a significant breaking change.